device.go

```diff
@@ -256,6 +246,46 @@ func nvdimmDeviceHandler(_ context.Context, device pb.Device, spec *pb.Spec, s *
 	return updateSpecDeviceList(device, spec, devIdx)
 }
 
+// device.Options should have one entry for each PCI device in the VFIO group
+// Each option should have the form "DDDD:BB:DD.F=NN/MM"
+//     DDDD:BB:DD.F is the device's PCI address in the host
+//     NN is the PCI slot of the device's bridge, MM is the PCI slot of the device
+func vfioDeviceHandler(ctx context.Context, device pb.Device, spec *pb.Spec, s *sandbox, devIdx devIndex) error {
+	fieldLogger := agentLog.WithField("host-dev", device.ContainerPath)
+
+	for _, opt := range device.Options {
+		tokens := strings.Split(opt, "=")
+		if len(tokens) != 2 {
+			return fmt.Errorf("Malformed VFIO option \"%s\"", opt)
+		}
+
+		hostBdf := tokens[0]
+		guestPCIPath := tokens[1]
```

Comment:
Given the vfio CI failure @devimc spotted in https://github.com/kata-containers/agent/pull/850#issuecomment-702141192, please could you check both these are `!= ""`.

Reply:
Um.. no.. or at least not right away.  I need to figure out why/how we're getting an empty string here, and probably fix the other side of this.